internal/decryption/kmip/client.go

```diff
@@ -0,0 +1,385 @@
+	UniqueIdentifier string
+}
+
+// GetRequest used for Get request operation.
+type GetRequest struct {
+	UniqueIdentifier kmip20.UniqueIdentifierValue
+}
+
+// GetResponse response of Get operation.
+type GetResponse struct {
+	ObjectType       kmip14.ObjectType
+	UniqueIdentifier string
+	SymmetricKey     kmip.SymmetricKey
+	PrivateKey       kmip.PrivateKey
+}
+
+// EncryptRequest used for Encrypt request operation.
+type EncryptRequest struct {
+	UniqueIdentifier kmip20.UniqueIdentifierValue
+	Data             []byte
+}
+
+// EncryptResponse response of Encrypt operation.
+type EncryptResponse struct {
+	UniqueIdentifier string
+	Data             []byte
+	IVCounterNonce   []byte
+}
+
+// DecryptRequest used for Decrypt request operation.
+type DecryptRequest struct {
+	UniqueIdentifier kmip20.UniqueIdentifierValue
+	Data             []byte
+	IVCounterNonce   []byte
+}
+
+// DecryptResponse response of Decrypt operation.
+type DecryptResponse struct {
+	UniqueIdentifier string
+	Data             []byte
+}
+
+// Symmetric key value.
+type KeyValue struct {
+	KeyMaterial []byte
+}
+
+// KMIP protocol version.
+type Version struct {
+	Major int
+	Minor int
+}
+
+var V10 = Version{Major: 1, Minor: 0} // first KMIP version
+var V12 = Version{Major: 1, Minor: 2} //nolint:gomnd // KMIP version that implemented encrypt / decrypt
+var V20 = Version{Major: 2, Minor: 0} //nolint:gomnd // KMIP major version change (create operation signature changed)
+
+var versions = map[Version]bool{V10: true, V12: true, V20: true}
+
+// cipherSuites is a list of enabled TLS 1.0–1.2 cipher suites.
+var cipherSuites = []uint16{
+	tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+	tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+	tls.TLS_RSA_WITH_AES_128_CBC_SHA256,
+}
+
+// Client client used to communicate with a KMIP speaking server.
+type Client struct {
+	version       Version
+	tlsConfig     tls.Config
+	requestHeader kmip.RequestHeader
+	ip            string
+	port          int
+}
+
+// Config structure used to configure a KMIP client.
+type Config struct {
+	Version           Version
+	IP                string
+	Port              int
+	Hostname          string
+	Username          string
+	Password          string
+	ClientPrivateKey  []byte
+	ClientCertificate []byte
+	RootCertificate   []byte
+}
+
+// NewClient creates a new KMIP client and initializes all the values required for establishing connection.
+func NewClient(config *Config) (*Client, error) {
+	if err := validate(config); err != nil {
+		return nil, err
+	}
+
+	rootCAs := x509.NewCertPool()
+	if !rootCAs.AppendCertsFromPEM(config.RootCertificate) {
+		return nil, errors.New("failed to load server's root certificate")
+	}
+
+	certificate, err := tls.X509KeyPair(config.ClientCertificate, config.ClientPrivateKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load client key and certificate: %s", err.Error())
```

Comment:
[non blocking/separate PR] you can do 
```suggestion
		return nil, fmt.Errorf("failed to load client key and certificate: %w", err)
```

Reply:
true :), will include it in my next PR (e2e for this code)